fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AstToStringBuilder.java

```diff
@@ -292,41 +310,227 @@ public String visitExportStatement(ExportStmt stmt, Void context) {
 
         @Override
         public String visitGrantRevokePrivilegeStatement(BaseGrantRevokePrivilegeStmt stmt, Void context) {
-            StringBuilder sb = new StringBuilder();
-            if (stmt instanceof GrantPrivilegeStmt) {
-                sb.append("GRANT ");
+            if (GlobalStateMgr.getCurrentState().isUsingNewPrivilege()) {
+                StringBuilder sb = new StringBuilder();
+                if (stmt instanceof GrantPrivilegeStmt) {
+                    sb.append("GRANT ");
+                } else {
+                    sb.append("REVOKE ");
+                }
+
+                List<String> privList = new ArrayList<>();
+                for (Map.Entry<String, Action> actionEntry : stmt.getObjectType().getActionMap().entrySet()) {
+                    if (stmt.getActionList().contains(actionEntry.getValue())) {
+                        privList.add(actionEntry.getValue().getName());
+                    }
+                }
+                sb.append(Joiner.on(", ").join(privList));
+                sb.append(" ON ");
+
+                switch (stmt.getObjectType()) {
+                    case TABLE:
+                    case VIEW:
+                    case MATERIALIZED_VIEW: {
+                        TablePEntryObject tablePEntryObject = (TablePEntryObject) stmt.getObjectList().get(0);
+                        if (tablePEntryObject.getDatabaseId() == TablePEntryObject.ALL_DATABASE_ID) {
+                            sb.append("ALL ").append(stmt.getObjectType().getPlural()).append(" IN ALL DATABASES");
+                        } else {
+                            Database database = GlobalStateMgr.getCurrentState().getDb(tablePEntryObject.getDatabaseId());
+                            if (tablePEntryObject.getTableId() == TablePEntryObject.ALL_TABLES_ID) {
+                                sb.append("ALL TABLES ");
+                                sb.append("IN DATABASE ").append(database.getFullName());
+                            } else {
+                                sb.append(stmt.getObjectType().name()).append(" ");
+
+                                Table table = database.getTable(tablePEntryObject.getTableId());
```

Comment:
what if database or table is null? currently the invalid privilege entry is cleaned lazily.